crates/ev-zkevm-types/src/programs/block.rs

```diff
@@ -458,3 +483,89 @@ impl BlockVerifier {
+        let new_state_len = bincode::serialize(&new_state).unwrap().len() as u64;
+        let output = BlockRangeExecOutput {
+            state_len_bytes: state_len.to_le_bytes(),
+            state: state.clone(),
+            new_state_len_bytes: new_state_len.to_le_bytes(),
+            new_state: new_state.clone(),
+        };
+
+        // Serialize
+        let serialized = bincode::serialize(&output).unwrap();
+
+        // Verify no length prefix was added - should be exactly 298 bytes
+        // (8 bytes for state_len + 141 bytes for state + 8 bytes for new_state_len + 141 bytes for new_state)
+        assert_eq!(
+            serialized.len(),
+            298,
+            "Serialized output should be exactly 298 bytes with no length prefix"
+        );
+
+        // Verify byte layout: first 8 bytes should be state_len_bytes
+        assert_eq!(&serialized[..8], state_len.to_le_bytes());
+
+        // Verify State serialization has no prefix (each State is 141 bytes)
+        let state_bytes = bincode::serialize(&state).unwrap();
+        let new_state_bytes = bincode::serialize(&new_state).unwrap();
+        assert_eq!(state_bytes.len(), 141, "State should serialize to 141 bytes");
+        assert_eq!(new_state_bytes.len(), 141, "State should serialize to 141 bytes");
+
+        // Verify the byte layout matches our expected format
+        assert_eq!(&serialized[8..149], state_bytes.as_slice());
+        assert_eq!(&serialized[149..157], new_state_len.to_le_bytes());
+        assert_eq!(&serialized[157..298], new_state_bytes.as_slice());
+
+        // Deserialize
+        let deserialized: BlockRangeExecOutput = bincode::deserialize(&serialized).unwrap();
+
+        // Verify round-trip
+        assert_eq!(output.state_len_bytes, deserialized.state_len_bytes);
+        assert_eq!(output.state.state_root, deserialized.state.state_root);
+        assert_eq!(
+            output.state.celestia_header_hash,
+            deserialized.state.celestia_header_hash
+        );
+        assert_eq!(output.state.celestia_height, deserialized.state.celestia_height);
+        assert_eq!(output.state.height, deserialized.state.height);
+        assert_eq!(output.state.namespace, deserialized.state.namespace);
+        assert_eq!(output.state.public_key, deserialized.state.public_key);
+        assert_eq!(output.new_state_len_bytes, deserialized.new_state_len_bytes);
+        assert_eq!(output.new_state.state_root, deserialized.new_state.state_root);
+        assert_eq!(
+            output.new_state.celestia_header_hash,
+            deserialized.new_state.celestia_header_hash
+        );
+        assert_eq!(output.new_state.celestia_height, deserialized.new_state.celestia_height);
+        assert_eq!(output.new_state.height, deserialized.new_state.height);
+        assert_eq!(output.new_state.namespace, deserialized.new_state.namespace);
+        assert_eq!(output.new_state.public_key, deserialized.new_state.public_key);
```

Comment:
I'm pretty sure that we can replace this with a single `assert_eq!()` if we derive `PartialEq, Eq` on both `State` and `BlockRangeExecOutput`